sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -237,4 +241,45 @@ class ScalaReflectionSuite extends SparkFunSuite {
     assert(anyTypes.forall(!_.isPrimitive))
     assert(anyTypes === Seq(classOf[java.lang.Object], classOf[java.lang.Object]))
   }
+
+  private def testThreadSafetyFor(name: String)(exec: () => Any) = {
+    test(s"thread safety of ${name}") {
+      for (_ <- 0 until 100) {
```

Comment:
`(0 until 100).foreach`? probably just a matter of taste but it's symmetrical with your inner loop then.
You can import `java.net.URLClassLoader`.
It doesn't really seem like you need a method here; it took a moment to see there was a test in here.

Maybe it's obvious to you but why do all these classes/methods need to be tested separately?

And is this locking still safe in 2.11?